File: Backend/barbershop/barbershop/Models/ResponseDTOs/CartDTO.cs
using barbershop.Models.Entitys;

namespace barbershop.Models.ResponseDTOs
{
    public class CartDTO
    {
        public long CartId { get; set; }

        public long? UserId { get; set; }

        public int? ProductId { get; set; }

        public int? SizeId { get; set; }
        public string? SizeName { get; set; } // Thêm thuộc tính SizeName
        public int? Quantity { get; set; }

        public bool? IsAvailable { get; set; }

        public  ProductDTO? ProductDTO { get; set; }
        public  ProductPriceDTO? ProductPriceDTO { get; set; } // Thêm thuộc tính ProductPriceDTO
        public  SizeDTO? SizeDTO { get; set; }

        public UserDTO? UserDTO { get; set; }

    }
}

File: Frontend/barbershop/src/types/ResponseDTOs/cartDTO.ts
import type { ProductDTO } from './productDTO';
import type { ProductPriceDTO } from './productPriceDTO';
import type { SizeDTO } from './sizeDTO';
import type { UserDTO } from './userDTO';

export type CartDTO = {
    cartId: number;
    userId: number | null;
    productId: number | null;
    sizeId: number | null;
    sizeName: string | null;
    quantity: number | null;
    isAvailable: boolean | null;
    productDTO: ProductDTO | null;
    productPriceDTO: ProductPriceDTO | null; // Thêm thuộc tính ProductPriceDTO
    sizeDTO: SizeDTO | null;
    userDTO: UserDTO | null;
};
